cmux.go

```diff
@@ -177,7 +177,8 @@ func (l muxListener) Accept() (net.Conn, error) {
 // MuxConn wraps a net.Conn and provides transparent sniffing of connection data.
 type MuxConn struct {
 	net.Conn
-	buf buffer
+	buf     bytes.Buffer
```

Comment:
Does this actually make things faster? I'm weary of cluttering the code with optimizations which are unsupported by benchmarks.

All of this is moot anyway, because I don't think this direction is better - you're packing more logic into one object, and there's no performance win here.